pkg/auth/handler/webapp/settings.go

```diff
@@ -60,6 +60,10 @@ type SettingsAuthorizationService interface {
 	Delete(a *oauth.Authorization) error
 }
 
+type SettingsSessionListingService interface {
+	FilterForDisplay(sessions []session.Session) ([]*model.Session, map[string]interface{}, error)
```

Comment:
Returning an extra map specific for authui seems strange. What about returning a new model?

```
type Session2 struct {
  *model.Session
  IsSSOEnable `json:"-"`
}
```

Reply:
Currently, we show the device icon for
- all the refresh tokens
- The IDP session that has SSO enabled refresh tokens

So the device icon is not for SSO enabled sessions only.
I amend the commit this way, see if it makes sense. fae1ec0